bundles/org.openhab.binding.sagercaster/src/main/java/org/openhab/binding/sagercaster/internal/SagerWeatherCaster.java

```diff
@@ -0,0 +1,400 @@
+            throw new IllegalStateException("Sager Forecaster properties file not found or not accessible");
+        }
+    }
+
+    public String[] getUsedDirections() {
+        return usedDirections;
+    }
+
+    public void setBearing(int newBearing, int oldBearing) {
+        int windEvol = sagerWindTrend(oldBearing, newBearing);
+        if ((windEvol != this.windEvolution) || (newBearing != currentBearing)) {
+            this.currentBearing = newBearing;
+            this.windEvolution = windEvol;
+            updatePrediction();
+        }
+    }
+
+    public void setPressure(double newPressure, double oldPressure) {
+        int newSagerPressure = sagerPressureLevel(newPressure);
+        int pressEvol = sagerPressureTrend(newPressure, oldPressure);
+        if ((pressEvol != this.pressureEvolution) || (newSagerPressure != sagerPressure)) {
+            this.sagerPressure = newSagerPressure;
+            this.pressureEvolution = pressEvol;
+            updatePrediction();
+        }
+    }
+
+    public void setCloudLevel(int cloudiness) {
+        this.cloudLevel = cloudiness;
+        sagerNubesUpdate();
+    }
+
+    public void setRaining(boolean raining) {
+        this.raining = raining;
+        sagerNubesUpdate();
+    }
+
+    public void setBeaufort(int beaufortIndex) {
+        if (currentBeaufort != beaufortIndex) {
+            currentBeaufort = beaufortIndex;
+            updatePrediction();
+        }
+    }
+
+    public int getBeaufort() {
+        return currentBeaufort;
+    }
+
+    public int getWindEvolution() {
+        return this.windEvolution;
+    }
+
+    public int getPressureEvolution() {
+        return this.pressureEvolution;
+    }
+
+    public void sagerNubesUpdate() {
+        int result;
+        if (!raining) {
+            if (cloudLevel > 80) {
+                result = 4; // overcast
+            } else if (cloudLevel > 50) {
+                result = 3; // mostly overcast
+            } else if (cloudLevel > 20) {
+                result = 2; // partly cloudy
+            } else {
+                result = 1; // clear
+            }
+        } else {
+            result = 5; // raining
+        }
+        if (result != nubes) {
+            nubes = result;
+            updatePrediction();
+        }
+    }
+
+    public static int sagerPressureLevel(double current) {
+        int result = 1;
+        if (current > 1029.46) {
+            result = 1;
+        } else if (current > 1019.3) {
+            result = 2;
+        } else if (current > 1012.53) {
+            result = 3;
+        } else if (current > 1005.76) {
+            result = 4;
+        } else if (current > 999) {
+            result = 5;
+        } else if (current > 988.8) {
+            result = 6;
+        } else if (current > 975.28) {
+            result = 7;
+        } else {
+            result = 8;
+        }
+        return result;
+    }
+
+    public static int sagerPressureTrend(double current, double historic) {
```

Comment:
It would be better for some of these methods to return an enum instead since it looks like each number has a discrete meaning from the comments and using enums would make the code more understandable and maintainable. Using enums where possible is a better coding practice.

Reply:
Sure it would be nicer but these number values are also the base of the Forecaster as recorded in the properties file - so I would prefer leave it as it is